src/core/lib/event_engine/posix_engine/ev_poll_posix.cc

```diff
@@ -341,10 +341,11 @@ void PollEventHandle::ShutdownHandle(absl::Status why) {
     // only shutdown once
     if (!is_shutdown_) {
       is_shutdown_ = true;
-      shutdown_error_ = why;
-      grpc_core::StatusSetInt(&shutdown_error_,
-                              grpc_core::StatusIntProperty::kRpcStatus,
-                              GRPC_STATUS_UNAVAILABLE);
+      shutdown_error_ = std::move(why);
+      grpc_core::StatusSetInt(
+          &shutdown_error_, grpc_core::StatusIntProperty::kRpcStatus,
+          absl::IsCancelled(shutdown_error_) ? GRPC_STATUS_CANCELLED
```

Comment:
This may have repercussions, but it's hard to tell. Be ready to roll back if necessary.